### DISCUSSION TITLE
SD card slow write SPI/Fatfs/stm32

### DISCUSSION THREAD

**Discussion by AlphaEmbed** [zephyrproject-rtos/zephyr#52398] [Q&A]
I want to write 1MB of data on SD card.
My data comes from ADC and I must write it directly to the SD card.
Main problem is I can't write data in Realtime operation.
Max speed I need is 2MB/s but I can get just about 3KB/s.
My hardware : 1) stm32f103c8 2)SPI SD card module

**Comment by AlphaEmbed** [comment]
I want to write 1MB of data on SD card.
My data comes from ADC and I must write it directly to the SD card.
Main problem is I can't write data in Realtime operation.
Max speed I need is 2MB/s but I can get just about 3KB/s. 
My hardware : 1) stm32f103c8  2)SPI SD card module

**Comment by EminYagmahan** [comment]
An SD card via SPI can reach a maximum data transfer rate of 10Mbit/s. This is 1.25 MB/s and therefore your expectation is not achievable. In combination with a file system the speed is reduced again. See the following issue #22906. There and in the following issues, there are more measurements.
